internal/dcache/metadata_manager/metadata_manager_impl.go

```diff
@@ -125,46 +161,209 @@ func GetClusterMap() ([]byte, *azcore.ETag, error) {
-	// Dummy implementation
-	return nil, nil
+	path := filepath.Join(m.mdRoot, "Objects", filePath)
+	// Get the metadata content from storage
+	data, err := m.storageCallback.GetBlobFromStorage(internal.ReadFileWithNameOptions{
+		Path: path,
+	})
+	if err != nil {
+		log.Debug("GetFile :: Failed to get metadata file content for file %s : %v", path, err)
+		return nil, err
+	}
+	// Unmarshal the JSON data into the Metadata struct
+	var metadata dcache.FileMetadata
+	err = json.Unmarshal(data, &metadata)
+	if err != nil {
+		log.Debug("GetFile :: Failed to unmarshal JSON data: %v", err)
+		return nil, err
+	}
+	// Return the Metadata struct
+	return &metadata, nil
 }
 
 // DeleteFile removes metadata for a file
 func (m *BlobMetadataManager) deleteFile(filePath string) error {
-	// Dummy implementation
-	return nil
+	path := filepath.Join(m.mdRoot, "Objects", filePath)
+	err := m.storageCallback.DeleteBlobInStorage(internal.DeleteFileOptions{
+		Name: path,
+	})
+	if err != nil {
+		if bloberror.HasCode(err, bloberror.BlobNotFound) {
+			log.Warn("DeleteFile :: DeleteBlobInStorage failed since blob %s is already deleted", path)
+			return nil
+		}
+	}
+	return err
 }
 
 // OpenFile increments the open count for a file and returns the updated count
 func (m *BlobMetadataManager) openFile(filePath string) (int64, error) {
-	// Dummy implementation
-
-	return 0, nil
+	path := filepath.Join(m.mdRoot, "Objects", filePath)
+	count, err := m.updateHandleCount(path, true)
+	if err != nil {
+		log.Err("OpenFile :: Failed to update file open count for path %s : %v", path, err)
+		return -1, err
+	}
+	return count, nil
 }
 
 // CloseFile decrements the open count for a file and returns the updated count
 func (m *BlobMetadataManager) closeFile(filePath string) (int64, error) {
-	// Dummy implementation
-	return 0, nil
+	path := filepath.Join(m.mdRoot, "Objects", filePath)
+	count, err := m.updateHandleCount(path, false)
+	if err != nil {
+		log.Err("CloseFile :: Failed to update file open count for path %s : %v", path, err)
+		return -1, err
+	}
+	return count, nil
+}
+
+func (m *BlobMetadataManager) updateHandleCount(path string, increment bool) (int64, error) {
+	const maxBackoff = 1 * time.Second // Maximum backoff time in seconds
+	backoff := 1 * time.Millisecond    // Initial backoff time in milliseconds
+	var openCount int
+	retryTime := time.Now()
+	for {
+		// Get the current handle count
+		attr, err := m.storageCallback.GetPropertiesFromStorage(internal.GetAttrOptions{
+			Name: path,
+		})
+		if err != nil {
+			log.Err("updateHandleCount :: Failed to get handle count for %s : %v", path, err)
+			return -1, err
+		}
+
+		// We never create file metadata blob w/o opencount property set.
+		if attr.Metadata["opencount"] == nil {
+			log.Err("updateHandleCount :: File metadata blob found w/o opencount property: %s", path)
+			return -1, fmt.Errorf("Opencount property not found in metadata for path %s. Issue needs to be debugged.", path)
```

Comment:
for such cases, log and return error, Akanksha has added a utility funciton . LogAndReturnError (or similar), check with here and use that for all such cases where you want to log and also return the same error string to the caller.